### PR TITLE
Switch configurations depending on detecting of auto-mining

### LINKED ISSUE #363: Switch configurations depending on detecting of auto-mining
Different default values for number of block confirmations should be set depending on whether an automining chain is detected.

If a user is overriding the values, these take precedence.

In the deploy function the `deployConfig` is merged with default values. At that point it should be determined if we are in an automining chain.

## Implementation details

* For hardhat and anvil: hardhat_getAutomine — Returns a boolean
* For ganache: web3_clientVersion and check if the string result includes "Ganache"

```js
if (isHardhatGetAutomineTrue) {
   return true;
}

if (isGanache) {
   return true; // This can be a false positive
}

return false;
```

We also want to add a validation on the passed in deployconfig that throws if the number of confirmations is below 1.

## TODO

- [x] Detect automining chain (local HH test node)
- [x] Rework defaults based on the check
- [x] Add validation check - 0 confirmations is an invalid value
- [x] Check the integration tests, unit tests and examples for config overrides that are no longer required

### PR DESCRIPTION
**Pull request by zoeyTM**
- [x] Detect automining chain (local HH test node)
- [x] Rework defaults based on the check
- [x] Add validation check - 0 confirmations is an invalid value
- [x] Check the integration tests, unit tests and examples for config overrides that are no longer required

fixes #363

### CODE REVIEW

File: packages/core/src/new-api/deploy.ts [hunk 1]
```diff
@@ -48,11 +49,14 @@ export async function deploy({
 
   const chainDispatcher = new ChainDispatcherImpl(buildAdaptersFrom(provider));
 
+  const isAutominedNetwork = await checkAutominedNetwork(provider);
```
Comment (kanej) [inline packages/core/src/new-api/deploy.ts:52]: I think we should push this check down into the `deploy` function of `deployer`.

WDYT?

File: packages/hardhat-plugin/test/config.ts [hunk 2]
```diff
@@ -1,39 +1,58 @@
...
+
+    it("should throw when given a `blockConfirmations` value less than 1", async function () {
+      const moduleDefinition = buildModule("FooModule", (m) => {
+        const foo = m.contract("Foo");
+
+        return { foo };
+      });
 
-    assert.deepStrictEqual(Object.keys(loadedOptions).sort(), configOptions);
+      await assert.isRejected(
```
Comment (kanej) [inline packages/hardhat-plugin/test/config.ts:52]: I think the deploy function for `useEphemeralIgnitionProject` isn't passing the config from `hre.config.ignition` into the deploy call here.

File: packages/core/src/new-api/deploy.ts (hunk 1 above)
Comment (kanej) [reply packages/core/src/new-api/deploy.ts:52]: After discussion we are happy with the current location. Async in a constructor would be bad.

File: packages/hardhat-plugin/test/config.ts (hunk 2 above)
Comment (kanej) [reply packages/hardhat-plugin/test/config.ts:52]: Done.

File: packages/core/src/new-api/internal/utils/check-automined-network.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+export async function checkAutominedNetwork(
+  provider: EIP1193Provider
+): Promise<boolean> {
+  const isHardhat = Boolean(
+    await provider.request({ method: "hardhat_getAutomine" })
+  );
+
+  if (isHardhat) {
+    return true;
+  }
```
Comment (alcuadrado) [inline packages/core/src/new-api/internal/utils/check-automined-network.ts:12]:
```suggestion
  try {
    const isHardhat = Boolean(
      await provider.request({ method: "hardhat_getAutomine" })
    );

    if (isHardhat) {
      return true;
    }
  } catch {
    // If this method failed we aren't using Hardhat Network nor Anvil, so we 
    // just continue with the next check.
  }
```

**Pull request merged**